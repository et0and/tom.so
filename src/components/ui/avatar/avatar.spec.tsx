import React from "react";
import { render } from "@testing-library/react";
import { Avatar, AvatarImage, AvatarFallback } from "./avatar";

describe("<Avatar/>", () => {
  it("renders Avatar correctly", () => {
    const { asFragment } = render(<Avatar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Avatar with AvatarImage correctly", () => {
    const { asFragment } = render(
      <Avatar>
        <AvatarImage src="test.jpg" alt="Test Avatar" />
      </Avatar>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Avatar with AvatarFallback correctly", () => {
    const { asFragment } = render(
      <Avatar>
        <AvatarFallback>TA</AvatarFallback>
      </Avatar>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Avatar with both AvatarImage and AvatarFallback", () => {
    const { asFragment } = render(
      <Avatar>
        <AvatarImage src="test.jpg" alt="Test Avatar" />
        <AvatarFallback>TA</AvatarFallback>
      </Avatar>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
