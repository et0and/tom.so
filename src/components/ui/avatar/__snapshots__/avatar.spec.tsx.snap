// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> renders Avatar correctly 1`] = `
<DocumentFragment>
  <span
    class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
  />
</DocumentFragment>
`;

exports[`<Avatar/> renders Avatar with AvatarFallback correctly 1`] = `
<DocumentFragment>
  <span
    class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
  >
    <span
      class="flex h-full w-full items-center justify-center rounded-full bg-gray-100 dark:bg-gray-800"
    >
      TA
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Avatar/> renders Avatar with AvatarImage correctly 1`] = `
<DocumentFragment>
  <span
    class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
  />
</DocumentFragment>
`;

exports[`<Avatar/> renders Avatar with both AvatarImage and AvatarFallback 1`] = `
<DocumentFragment>
  <span
    class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
  >
    <span
      class="flex h-full w-full items-center justify-center rounded-full bg-gray-100 dark:bg-gray-800"
    >
      TA
    </span>
  </span>
</DocumentFragment>
`;
